### ISSUE TITLE
[JIT] Generalize variadic output logic for min, max and split

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#8528]
Operator-level tests failed:

```
======================================================================
FAIL: test_max_scalar_elementwise_broadcast_rhs (__main__.TestAutogradGenerated)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_autograd.py", line 3665, in do_test
    check(name)
  File "test/test_autograd.py", line 3592, in check
    fn, (self_variable,) + args_variable)
  File "test/test_autograd.py", line 3499, in check_against_reference
    self.assertEqual(outputs, outputs_test)
  File "/data/users/zdevito/pytorch/test/common.py", line 312, in assertEqual
    super(TestCase, self).assertEqual(len(x), len(y), message)
AssertionError: 5 != 2

======================================================================
FAIL: test_min_scalar_elementwise_broadcast_rhs (__main__.TestAutogradGenerated)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_autograd.py", line 3665, in do_test
    check(name)
  File "test/test_autograd.py", line 3592, in check
    fn, (self_variable,) + args_variable)
  File "test/test_autograd.py", line 3499, in check_against_reference
    self.assertEqual(outputs, outputs_test)
  File "/data/users/zdevito/pytorch/test/common.py", line 312, in assertEqual
    super(TestCase, self).assertEqual(len(x), len(y), message)
AssertionError: 5 != 2

======================================================================
FAIL: test_split_dim (__main__.TestAutogradGenerated)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_autograd.py", line 3665, in do_test
    check(name)
  File "test/test_autograd.py", line 3587, in check
    check_against_reference(self, create_traced_fn(fn), fn, (self_variable,) + args_variable)
  File "test/test_autograd.py", line 3499, in check_against_reference
    self.assertEqual(outputs, outputs_test)
  File "/data/users/zdevito/pytorch/test/common.py", line 312, in assertEqual
    super(TestCase, self).assertEqual(len(x), len(y), message)
AssertionError: 1 != 5

======================================================================
FAIL: test_split_dim_neg0 (__main__.TestAutogradGenerated)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_autograd.py", line 3665, in do_test
    check(name)
  File "test/test_autograd.py", line 3587, in check
    check_against_reference(self, create_traced_fn(fn), fn, (self_variable,) + args_variable)
  File "test/test_autograd.py", line 3499, in check_against_reference
    self.assertEqual(outputs, outputs_test)
  File "/data/users/zdevito/pytorch/test/common.py", line 312, in assertEqual
    super(TestCase, self).assertEqual(len(x), len(y), message)
AssertionError: 1 != 5
```

Affected tests:
- `test_split`
- `test_max_scalar_elementwise_broadcast_rhs`
- `test_min_scalar_elementwise_broadcast_rhs`
- `test_split_dim`
- `test_split_dim_neg0`

**Comment by jamesr66a** [comment]
@zou3519 I believe this will be fixed with your changes. The schema that's being resolved here (at least for `test_{min,max}` is one that returns a tuple in the case that we pass in a scalar `dim` parameter. I believe we're matching the schema because we're treating the second tensor operand as `dim` and emitting that op with 2 outputs.

**Comment by jamesr66a** [comment]
I think we'll also need to generalize this logic for aten::min, aten::max and aten::split as well https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/script/compiler.cpp#L494

**Comment by wanchaol** [comment]
Note: for the min, max operator, we are matching the wrong schema when we are passing in two tensors (treating the second tensor operand as dim), this will depends on the Richards' changes as well.

**Comment by wanchaol** [comment]
Note: for split in [tensor](https://github.com/pytorch/pytorch/blob/master/torch/tensor.py#L264), we have split and split_with_sizes depending on the type of split_size, currently in script we could not be able to match the schema with split_with_sizes when providing an intList. This applies with other functions like index_add, index_fill, etc. We need to find a generic way to support them after @zou3519 and @zdevito changes.

**Comment by zou3519** [comment]
test_max_scalar_elementwise_broadcast_rhs
test_min_scalar_elementwise_broadcast_rhs
are now fixed; the split tests aren't.

**Comment by wanchaol** [comment]
Split need to wait until we have mutable list implemented, as it requires us to know the shape of the input tensor which we could only know it during runtime.

**Comment by driazati** [comment]
Broadcasts fixed by #8687, splits fixed by #11051 and tests removed #11403
